src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        long start = System.nanoTime();
+        final SimpleDateFormat formatter = new SimpleDateFormat("dd.MM.y");
+        int[] salaries = new int[names.length];
+        try {
+            Date dateFromDataType = formatter.parse(dateFrom);
+            Date dateToDataType = formatter.parse(dateTo);
+
+            for (int i = 0; i < data.length; i++) {
+                String[] separateData = data[i].split(" ");
+                if ((formatter.parse(separateData[0]).after(dateFromDataType)
+                        || formatter.parse(separateData[0]).equals(dateFromDataType))
+                        && (formatter.parse(separateData[0]).before(dateToDataType)
+                        || formatter.parse(separateData[0]).equals(dateToDataType))) {
```

Comment:
can we use formatter.parse(separateData[0]) only ones before if(not to do the same operation several times)